### PR TITLE
Resolve CoreML silent CPU fallback for external-data ONNX models on macOS

### PR DESCRIPTION
**Pull request by mario-rei**
🎯 What problem does this PR solve?

- CoreML EP in ORT ≤1.24 crashes with !model_path.empty() was false on models with .onnx_data sidecars — ModelBuilder::AddConstant() passes an empty path to Initializer. 
- useGPU: true on macOS silently falls back to CPU with no user-visible error

📝 How does it solve it?

- resolveCoremlModelPath() in index.js swaps external-data model paths to single-file *_coreml.onnx variants when useGPU: true on macOS (applied in both _load() and reload())
- Models without a _coreml variant are left unchanged — CPU fallback works as before
- scripts/prepare-coreml-models.py generates the single-file variants from external-data models

🧪 How was it tested?

- Unit tests for resolveCoremlModelPath edge cases (external data present/absent, _coreml variant present/absent, falsy inputs)
- Integration tests verifying _load() and reload() resolve paths correctly with useGPU: true on darwin
- Manual verification: reproduced !model_path.empty() error with original paths, confirmed fix loads CoreML session successfully with _coreml variant

### CODE REVIEW

File: packages/qvac-lib-infer-onnx-tts/index.js [hunk 1]
```diff
@@ -12,6 +13,28 @@ const { TTSInterface } = require('./tts')
...
+ *
+ * When useGPU is true on macOS, swap each external-data .onnx path for a
+ * single-file variant (<basename>_coreml.onnx) if one exists on disk.
+ * Models without external data or without a _coreml variant are left as-is
+ * (the C++ layer's try/catch will fall back to CPU for those).
+ */
+function resolveCoremlModelPath (onnxPath) {
+  if (!onnxPath) return onnxPath
+  const dataFile = onnxPath + '_data'
+  try { fs.statSync(dataFile) } catch (_) { return onnxPath }
```
Comment (sharmaraju352) [inline packages/qvac-lib-infer-onnx-tts/index.js]: let's not swallow erros

File: packages/qvac-lib-infer-onnx-tts/index.js [hunk 2]
```diff
@@ -12,6 +13,28 @@ const { TTSInterface } = require('./tts')
...
+
+  const dir = path.dirname(onnxPath)
+  const base = path.basename(onnxPath, '.onnx')
+  const candidate = path.join(dir, base + '_coreml.onnx')
+  try { fs.statSync(candidate) } catch (_) { return onnxPath }
```
Comment (sharmaraju352) [inline packages/qvac-lib-infer-onnx-tts/index.js]: same here
